### DISCUSSION TITLE
With FCOS 34 systemd-resolved handles the content of /etc/resolv.com (via symbolic links) with regard to empty domains.  This breaks OKD 4.7 and 4.8 in specific cases

### DISCUSSION THREAD

**Discussion by fortinj66** [okd-project/okd#719] [General]
Prior to FCOS 34, systemd-resolved would create a /etc/resolv.conf without a `search` line if no search domains were defined, such as when IP addresses are assigned via kernel args...  Due to glibc magic, a search domain would be derived if the hostname is fully qualified.  

OKD works fine in this case.

However, with FCOS 34, this behavior has changed and now systemd-resolved will add a line 
`search .` 
to /etc/resolv.conf.

see https://github.com/systemd/systemd/pull/17201

This has the unfortunate effect breaking internal DNS resolution within the OKD cluster.  DNS lookups for names such as `image-registry.openshift-image-registry.svc` fail.  see https://github.com/openshift/okd/discussions/694

We have a potential workaround in https://github.com/openshift/okd-machine-os/pull/159.

The question is what is the best long-term solution for this issue.  I don't know all the internals working of the cluster DNS resolution process so I'm not sure what the next steps need to be. 

**Possibilities:**

- make the internal DNS resolution work out of the box with the new systemd-resolved behavior
- request rollback of the systemd pull request above
- request that they provide a flag to determine the behavior needed
- some  other solution

**Comment by fortinj66** [comment]
nightly `registry.ci.openshift.org/origin/release:4.7.0-0.okd-2021-06-29-103107` has the workaround for this issue...

**Comment by fortinj66** [comment]
Interesting article about systemd-resolved...

https://fedoramagazine.org/systemd-resolved-introduction-to-split-dns/

**Comment by LorbusChris** [comment]
ref: https://bugzilla.redhat.com/show_bug.cgi?id=1874419
